packages/open-payments/src/client/index.ts

```diff
@@ -166,40 +169,49 @@ const createAuthenticatedClientDeps = async ({
 }:
   | CreateAuthenticatedClientArgs
   | CreateAuthenticatedClientWithReqInterceptorArgs): Promise<AuthenticatedClientDeps> => {
-  const logger = args?.logger ?? createLogger({ name: 'Open Payments Client' })
+  const logger =
+    args?.logger ??
+    createLogger({ name: 'Open Payments Client', level: 'silent' })
   if (args.logLevel) {
     logger.level = args.logLevel
   }
 
-  let privateKey: KeyObject | undefined
-  try {
-    privateKey = parseKey(args)
-  } catch (error) {
-    const errorMessage =
-      'Could not load private key when creating Open Payments client'
-    const description = error instanceof Error ? error.message : 'Unknown error'
-
-    logger.error({ description }, errorMessage)
-
-    throw new OpenPaymentsClientError(errorMessage, {
-      description
-    })
-  }
-
   let httpClient: HttpClient
 
   if ('authenticatedRequestInterceptor' in args) {
     httpClient = await createHttpClient({
+      logger,
       requestTimeoutMs:
         args?.requestTimeoutMs ?? config.DEFAULT_REQUEST_TIMEOUT_MS,
-      authenticatedRequestInterceptor: args.authenticatedRequestInterceptor
+      requestSigningArgs: {
+        authenticatedRequestInterceptor: args.authenticatedRequestInterceptor
+      }
     })
   } else {
+    let privateKey: KeyObject
+    try {
+      privateKey = parseKey({ logger }, args.privateKey)
+    } catch (error) {
+      const errorMessage =
+        'Could not load private key when creating authenticated client'
+      const description =
+        error instanceof Error ? error.message : 'Unknown error'
+
+      logger.error({ description }, errorMessage)
+
+      throw new OpenPaymentsClientError(errorMessage, {
+        description
+      })
```

Comment:
Might be worthwhile to pass `Error.cause` for better debugging sometime.

Reply:
Yes, that would be a nice improvement to the errors. I think this is something to tackle when updating Typescript in the repo, as we currently don't have the updated types for `Error` class that includes `cause`. We can do it then such that we do it properly instead of using `ts-ignore`

Reply:
Aside: On updating TypeScript, I was updating `@apidevtools/json-schema-ref-parser` today (due to some node-fetch error it apparently causes on Windows), but then it required updating pretty much all dependencies, from openapi-typescript to TypeScript 😄.
Should do that sometime - nearly all dependencies pretty out of date

Reply:
> Should do that sometime - nearly all dependencies pretty out of date

Yes, maybe in a next sprint I can go through it 😅